stores/directory/local.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod, abstractstaticmethod
+from pathlib import Path
+from typing import List
+
+from stores.directory.base import BaseWritableSingleDirectoryStore
+
+
+class LocalDirectoryStore(BaseWritableSingleDirectoryStore):
+
+    local_path: Path
+
+    def __init__(self, local_dir):
+        # Takes a path or a string representing a path as input
+
+        # If it is not a path, pathify it
+        if type(local_dir) != Path:
+            try:
+                local_dir = Path(local_dir)
+            except:
+                raise TypeError(f"Given path {local_dir} is not a path and cannot be converted to a path.")
+
+        # Make sure it exists
+        assert local_dir.exists(), "Given path does not exist as a local directory."
```

Comment:
`f"Given path {local_dir.absolute()} does not exist."` - really spell out where the problem is.

Raise a good point actually, can you do two assertions please?

1. - assert it exists
2. - once we've asserted that assert its a directory, im 90% sure there's a pathlib method for that.